src/api/app/components/notification_notifiable_link_component.rb

```diff
@@ -138,6 +139,7 @@ def notifiable_link_path
   end
   # rubocop:enable Metrics/CyclomaticComplexity
   # rubocop:enable Metrics/PerceivedComplexity
+  # rubocop:enable ViewComponent/AvoidGlobalState
```

Comment:
I think ruby believes we're consuming a `@notification` instance variable coming from the controller, instead of being an instance variable set by the component. This can be fixed by using `attr_reader :notification` and replacing those `@notification` with `notification`.

Reply:
Comes from https://github.com/openSUSE/open-build-service/blob/master/src/api/lib/rubocop/cop/view_component/avoid_global_state.rb

Reply:
@danidoni actually, it doesn't. It complains about using `User` class (it should complain also about the entire Component code where the `@notification` instance is used intensively otherwise). It would like to receive the two variables I need by using `User` class as parameters in the `initialize`, but I actually would like to avoid that.

For now I am extracting the complained code into [two methods](https://github.com/openSUSE/open-build-service/pull/15669/files#diff-adcdf476178524b64ee66f1b607eb3359a059694756a060ea2082443ac9ffba9R209-R220) to localize it better and not involving the rest. I'll research about the complain a bit further then.

Reply:
Moving the logic to the `helpers` does the trick to rubocop and becomes the workaround https://github.com/openSUSE/open-build-service/pull/15669/commits/d857eb5c9c35be7afbaa06740ab3c005d8300647

Reply:
You are right. We're querying the database here, inside the view component, and if we follow the best practices, we should avoid it. We could pass the user triggering the event and the current user as parameters of the component. It should not be hard.

Reply:
Please have a look at the whole result: `event_user` is a _property_ of `Notification` itself, and the `current_user` is passed as a parameter along the codepath. It adds a bit more of a _here and there_ but it is compliant now.

Reply:
Alright, now everything makes sense